content/en/users/tutorials/adhoc/data-redundancy/_index.md

```diff
@@ -0,0 +1,187 @@
+> https://github.com/restic/restic
+
+---
+
+## Solution 1: Redundant OpenStack Setup
+_**Using Rsync and Snapshot Replication**_
+
+### Overview
+
+This solution synchronizes virtual machine (VM) snapshots between two OpenStack instances, ensuring high availability and data redundancy.
+
+### Prerequisites
+
+- A virtual organization with access to two OpenStack sites (Source and Destination sites)
+- A migration instance on both sites
+- SSH access between sites
+- OpenStack CLI tools installed
+- Sufficient storage capacity
+
+### Step 1: Configure SSH for Passwordless Authentication
+
+On the source site:
+
+```sh
+ssh-keygen -t rsa
+ssh-copy-id user@destination_host
+```
+
+### Step 2: Snapshot Replication
+
+#### 2.1 Create a Snapshot on the Source
+
+```sh
+openstack server snapshot create --name instance_snapshot INSTANCE_ID
+```
+
+#### 2.2 Transfer the Snapshot to the Destination
+
+```sh
+openstack image list | grep instance_snapshot
+openstack image save --file instance_snapshot.img SNAPSHOT_ID
+scp instance_snapshot.img user@destination_host:/tmp/
+```
+
+#### 2.3 Import Snapshot on Destination
+
+```sh
+openstack image create --file /tmp/instance_snapshot.img --disk-format qcow2 --container-format bare instance_snapshot
+```
+
+### Step 3: Automate Snapshot Replication
+
+Edit the cron job:
+
+```sh
+crontab -e
+```
+
+Add:
+
+```sh
+0 2 * * * openstack server snapshot create --name instance_snapshot-$(date +\%F) INSTANCE_ID
+0 3 * * * openstack image save --file /tmp/instance_snapshot-$(date +\%F).img $(openstack image list | grep instance_snapshot-$(date +\%F) | awk '{print $2}')
+0 4 * * * scp /tmp/instance_snapshot-$(date +\%F).img user@destination_host:/tmp/
+0 5 * * * ssh user@destination_host 'openstack image create --file /tmp/instance_snapshot-$(date +\%F).img --disk-format qcow2 --container-format bare instance_snapshot-$(date +\%F)'
+```
+
+### Step 4: Testing Failover
+
+On the destination site:
+
+```sh
+openstack server create --flavor FLAVOR_ID --image instance_snapshot --network NETWORK_ID new_instance
+```
+
+### Results
+
+By utilizing rsync and snapshot replication, VM state consistency is ensured between sites, improving failover readiness.
+
+---
+## Solution 2: Redundant Object Storage
+_**with MinIO**_
+
+### Overview
+
+MinIO is an open-source, high-performance object storage solution compatible with Amazon S3. It enables redundancy and data replication across multiple sites, making it a good alternative for OpenStack Swift or AWS S3 in a private or federated cloud.
+
+### Deployment in FedCloud
+
+MinIO is available in FedCloud with different deployment options:
+
+- **Preconfigured Virtual Machines** using Infrastructure Manager.
+- **Helm chart deployment in Kubernetes**, with CESNET's Rancher enabling easy setup with just a few clicks.
```

Comment:
Instead of CESNET's Rancher, I would refer to Cloud Container Compute.

Reply:
If that’s the case, I would prefer to indicate both names. In my opinion, EGI’s branding of well-known products does not enhance clarity.